### ISSUE TITLE
[오예은] Section2 solo project

### ISSUE THREAD

**Issue by dsdoris** [codestates/blog-ds#233]
[노트북보기](https://colab.research.google.com/drive/1bVMob4zF2c-yinXu0IGsT6zXKp2Alow5?usp=sharing)

- 분석 목표: 고객의 숙박 예약 취소 예측하기
- 분석 데이터: 호텔 예약 정보 (고객유형, 예약상태, 도착정보, 이용서비스종류, 예약경로 등 32개의 특성)
- 데이터 출처: Kaggle- Hotel booking demand(https://www.kaggle.com/jessemostipak/hotel-booking-demand)

## 전처리
분석목표인 취소 예측과 관련하여 아래 질문들을 가지고 데이터를 살펴보고 다듬기로 했다.

Q. 예약 취소에 아이들 인원이 영향을 미칠까?
Q. 예전에 취소했던 이력이 있는 사람이 그렇지 않은 사람보다 취소할 확률이 더 높을까?
Q. 호텔에 요청 사항이 있는 고객일수록 취소를 못할까?
Q. 1박을 초과하여 머무는 사람이 1박으로 머무는 사람보다 취소할 확률이 낮을까?

![image](https://user-images.githubusercontent.com/69617800/98362572-d2344900-2070-11eb-9008-b7b5dc94c11d.png)
> 그래프를 보면 아이들이 몇명인지와 취소여부는 관계가 거의 없을 것 같다.

![image](https://user-images.githubusercontent.com/69617800/98363145-d7de5e80-2071-11eb-9213-bd17295bd216.png)
> 그래프를 보면 예전에 취소했던 이력이 14회 이상일때는 이번 예약 때도 취소하는 것으로 나타났다.

### Target
- is_canceled (취소여부)
### Baseline (비율)
- 취소한 고객 37.0%, 취소하지 않은 고객 63.0%로 취소하지 않는 고객이 많은 쪽으로 치우쳐져있다.
### Classifier
- XG boost Classifier (n_estimators=1000, max_depth=7, learning_rate=0.2)
### Metrics
- 베이스라인의 두 분류간 차지하는 비중의 차가 크지 않다고 생각하여 Accuracy(정확도)를 평가지표로 사용하기로 했다.
- 정확도를 비롯한 모델의 평가지표는 아래와 같다.
![image](https://user-images.githubusercontent.com/69617800/98363419-4ae7d500-2072-11eb-9391-270e0f154771.png)

### Permutation Importance (25 features)
![image](https://user-images.githubusercontent.com/69617800/98363380-3d324f80-2072-11eb-8181-5d4f6a7c790d.png)

### Shap value로 모델의 예측 살펴보기
- 고객1
![image](https://user-images.githubusercontent.com/69617800/98363575-86829f00-2072-11eb-9a7a-37e953c61ea1.png)

-고객2
![image](https://user-images.githubusercontent.com/69617800/98363652-a7e38b00-2072-11eb-895b-c30d8fda586b.png)

#### 추후 모델을 이용하여
- 예약을 취소할 것으로 예측되는 고객이 실제로 예약을 취소하기 전에 고객이 원하는 대안을 제시함으로 취소를 방어하는 데 활용
- 이후 주로 예약을 취소하는 고객들의 공통적인 특성을 파악하여 서비스 개선에 기여할 수 있음
